VideoApp/VideoApp/Video/Participants/LocalParticipant.swift

```diff
@@ -74,7 +73,28 @@ class LocalParticipant: NSObject, Participant {
         }
     }
     var participant: TwilioVideo.LocalParticipant? {
-        didSet { participant?.delegate = self }
+        didSet {
+            guard let participant = participant else { return }
+            
+            participant.delegate = self
+
+            // Sync tracks in case user made changes while connecting to room
+            participant.localVideoTracks.compactMap { $0.localTrack }.filter { $0 !== cameraManager?.track.track }.forEach {
+                participant.unpublishVideoTrack($0)
+            }
+
+            if let cameraTrack = cameraManager?.track.track, !participant.localVideoTracks.contains(where: { $0.localTrack === cameraTrack }) {
+                participant.publishCameraTrack(cameraTrack)
+            }
+
+            participant.localAudioTracks.compactMap { $0.localTrack }.filter { $0 !== micTrack }.forEach {
+                participant.unpublishAudioTrack($0)
+            }
+
+            if let micTrack = micTrack, !participant.localAudioTracks.contains(where: { $0.localTrack === micTrack }) {
+                participant.publishAudioTrack(micTrack)
+            }
```

Comment:
There are other ways to solve this problem but the UX is not as good. The UI could disable buttons while connecting. Or maybe the app could wait until connected to the room to publish tracks. 

I did look into writing unit tests for this change but I think this file (and likely most of the SDK integration layer) cannot be unit tested because it is impossible to mock things like `TwilioVideo.LocalParticipant`. I can't mock it because it is not a protocol and `init` is not public. This is pretty much always an issue with SDKs so it is not surprising. This is another reason to have the SDK integration layer. The SDK integration layer isolates the code that is difficult to unit test. And then the higher layers like the transform layer don't have unit test problems because SDK integration layer is easy to mock. I suspect I will need to rely on UI tests to test the SDK integration layer. Using more protocols in public SDK interface would help but not sure making the SDK interface testable is a priority. Or maybe someday there will be better ways to mock in Swift.

Reply:
Actually regarding `TwilioVideo.LocalParticipant` I think I can work around that test problem. I can retroactive model it (create a protocol and make it conform to it) perhaps. I might take another crack at that.

Reply:
I'm gonna merge this and look into unit tests separately because I think unit tests will sort of be a different PR decision. It's going to be interesting because it will be the first SDK integration code with unit tests.

Reply:
Ok so I got further with unit testing but ultimately hit a roadblock. 

I was able to retroactive model `TwilioVideo.LocalParticipant` without issue like this:
```swift
import TwilioVideo

protocol TwilioVideoLocalParticipantProtocol: AnyObject {
    var networkQualityLevel: NetworkQualityLevel { get }
    var delegate: LocalParticipantDelegate? { get set }
    var localAudioTracks: [LocalAudioTrackPublication] { get }
    var localVideoTracks: [LocalVideoTrackPublication] { get }
    @discardableResult func publishAudioTrack(_ track: LocalAudioTrack) -> Bool
    @discardableResult func publishVideoTrack(_ track: TwilioVideo.LocalVideoTrack, publicationOptions options: LocalTrackPublicationOptions) -> Bool
    @discardableResult func unpublishAudioTrack(_ track: LocalAudioTrack) -> Bool
    @discardableResult func unpublishVideoTrack(_ track: TwilioVideo.LocalVideoTrack) -> Bool
}

extension TwilioVideo.LocalParticipant: TwilioVideoLocalParticipantProtocol { }
```
But: 

1. Probably not a big deal but this does make the code a little harder to follow especially if someone is not familiar with this technique.
2. The last straw was not being able to create `LocalAudioTrackPublication` or `LocalVideoTrackPublication`. I can't test the changes in this PR because the code under test does things like `participant.localVideoTracks` and I have no way to create tracks for my mock participant.

I think the only way around this (without SDK changes) would be to create another class between `VideoApp.LocalParticipant` and `TwilioVideo.LocalParticipant` that is just a testable facade and includes no real logic. This can make a lot of sense if an SDK interface is small but the TwilioVideo interface is so large this does not seem practical. 

And even if I didn't run into this roadblock I'm not sure it would be a good idea to do all this setup for tests because you just don't know what surprises there will be in the future.

So I'm back to thinking I cannot mock the SDK and for the SDK integration layer that means either don't test it at all, use UI tests without mocks (not unusual), or try unit or integration tests without mocks (pretty unusual). I need to think about which of those options makes the most sense and welcome feedback. But I think conventional unit tests for SDK integration layer are off the table.